igvc_gazebo/nodes/magnetometer/main.cpp

```diff
@@ -0,0 +1,30 @@
+#include <ros/ros.h>
+#include <geometry_msgs/Vector3Stamped.h>
+#include <sensor_msgs/MagneticField.h>
+
+ros::Publisher mag_field_pub_;
+
+void magCallback(const geometry_msgs::Vector3Stamped& msg)
+{
+  sensor_msgs::MagneticField magnet_msg;
+  magnet_msg.header.seq = msg.header.seq;
+  magnet_msg.header.stamp = msg.header.stamp;
+  magnet_msg.header.frame_id = msg.header.frame_id;
+  magnet_msg.magnetic_field.x = msg.vector.x;
+  magnet_msg.magnetic_field.y = msg.vector.y;
+  magnet_msg.magnetic_field.z = msg.vector.z;
+  magnet_msg.magnetic_field_covariance = { 1e-6, 0, 0, 0, 1e-6, 0, 0, 0, 1e-6 };
```

Comment:
I feel conflicted about having our probably incorrect covariance in multiple places, but oh well.